app/models/user.rb

```diff
@@ -577,6 +577,36 @@ def authenticate_by_personal_access_token(token)
 
   private
 
+  def allowed_controller_actions
```

Comment:
Definitely not a good name. It sounds pretty general, but it really is not.
Otherwise I do like the refactoring, but it's now unrelated to the issue, so it should be separate PR